src/journal/journald-audit.c

```diff
@@ -264,7 +264,8 @@ static int map_all_fields(
                 const char *v;
 
                 if (*n >= m) {
-                        log_debug(
+                        log_ratelimit_debug(
+                                JOURNALD_LOG_RATELIMIT,
```

Comment:
Hmm, is it really better to rate-limit each message site individually? I wonder if having a single RateLimit struct for all internal journal messages wouldn't be better? It should then have a much higher `burst`, for the cases where we do some operation and produce a bunch of messages. One LogRateLimit is 32 bytes, and I wonder if it is useful to have hundreds of those variables… (This is really an unbiased question. I'm not convinced either way.)

Reply:
The problem with that approach is that one noisy log message can drown out other less noisy log messages. Doing per message rate limiting makes sure that each message actually gets logged.